### PR TITLE
Add collections-related test suite

### LINKED ISSUE #4: Add test suite for collections-related methods
The package should have, at a minimum, unit tests set up for `WriteFreelyClient`'s public API.

This issue is therefore blocked by #3 — implementing dependency injection for `URLSessionDataTask`.

### PR DESCRIPTION
**Pull request by AngeloStavrow**
This PR closes #4, adding tests for collection-related endpoints.

### CODE REVIEW

File: Package.swift [hunk 1]
```diff
@@ -27,6 +27,10 @@ let package = Package(
             dependencies: []),
         .testTarget(
             name: "WriteFreelyTests",
-            dependencies: ["WriteFreely"]),
+            dependencies: ["WriteFreely"],
+            resources: [
+                .process("Resources")
+            ]
```
Comment (AngeloStavrow) [inline Package.swift:33]: This lets us use the test JSON files under `Tests/WriteFreelyTests/Resources`.

File: Tests/WriteFreelyTests/TestUtils.swift [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    func resume() {
+        resumeWasCalled = true
+    }
+}
+
+final class MockURLSession: URLSessionProtocol {
+    var nextDataTask = MockURLSessionDataTask()
+    var nextData: Data?
+    var nextError: Error?
+    var expectedStatusCode: Int = 200
```
Comment (AngeloStavrow) [inline Tests/WriteFreelyTests/TestUtils.swift:17]: This lets us set our expected status code explicitly in our tests — not everything expects a `200 OK`!

File: Tests/WriteFreelyTests/TestUtils.swift [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+        completionHandler(
+            nextData,
+            successURLResponse(url: (request.url ?? URL(string: "https://example.com")!)),
+            nextError
+        )
+        return nextDataTask
+    }
+
+    @discardableResult
+    func setData(resource: String, fileExt: String, for target: XCTestCase) throws -> URL? {
```
Comment (AngeloStavrow) [inline Tests/WriteFreelyTests/TestUtils.swift:41]: We use this to set the response JSON from a file in our `Resources` path.

File: Tests/WriteFreelyTests/WriteFreelyClientCollectionTests.swift [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+
+final class WriteFreelyClientCollectionTests: XCTestCase {
+    static var allTests = [
+        ("testCreateCollection_WithValidCollectionData_CreatesNewWFCollection", testCreateCollection_WithValidCollectionData_CreatesNewWFCollection),
+        ("testCreateCollection_WithInvalidCollectionData_ReturnsBadRequestError", testCreateCollection_WithInvalidCollectionData_ReturnsBadRequestError),
+        ("testGetCollection_WithValidCollectionData_RetrievesCollectionMetadata", testGetCollection_WithValidCollectionData_RetrievesCollectionMetadata),
+        ("testGetCollection_WithInvalidCollectionData_ReturnsInvalidDataError", testGetCollection_WithInvalidCollectionData_ReturnsInvalidDataError),
+        ("testDeleteCollection_WithValidCollectionAlias_ReturnsTrue", testDeleteCollection_WithValidCollectionAlias_ReturnsTrue),
+        ("testDeleteCollection_WithInvalidCollectionAlias_ReturnsInvalidResponseError", testDeleteCollection_WithInvalidCollectionAlias_ReturnsInvalidResponseError)
+    ]
```
Comment (AngeloStavrow) [inline Tests/WriteFreelyTests/WriteFreelyClientCollectionTests.swift:12]: We focus here on the `createCollection`, `getCollection`, and `deleteCollection` methods on WFClient as "collection"-related methods.

The REST API includes things like getting a post from a collection, for example, but in the client we make that a "post"-related method as an overload on `getPost`.

**Review by thebaer — APPROVED** [review]
This looks good to me!

**Pull request merged**